### PR TITLE
IEEE802.15.4: Remove dyn, change to template

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

Remove `dyn` from RadioDriver.

### Testing Strategy

travis

### TODO or Help Wanted

n/a

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

**Review by lschuermann — APPROVED** [review]

File: boards/components/src/ieee802154.rs [hunk 1]
```diff
@@ -95,10 +97,39 @@ macro_rules! ieee802154_component_static {
             >
         );
 
-        let mux_mac = kernel::static_buf!(capsules_extra::ieee802154::virtual_mac::MuxMac<'static>);
-        let mac_user =
-            kernel::static_buf!(capsules_extra::ieee802154::virtual_mac::MacUser<'static>);
-        let radio_driver = kernel::static_buf!(capsules_extra::ieee802154::RadioDriver<'static>);
+        let mux_mac = kernel::static_buf!(
+            capsules_extra::ieee802154::virtual_mac::MuxMac<
```
Comment (hudson-ayers) [inline boards/components/src/ieee802154.rs:101]: This file would probably become a lot more readable with a
```rust
use capsules_extra::ieee802154::virtual_mac;
use capsules_extra::ieee802154::framer;
```
etc.,
but I don't think we need to block this PR on it.

**Review by hudson-ayers — DISMISSED** [review]
This is a nice change

**Comment by bradjc** [comment]
Can we merge this?

**Review by lschuermann — APPROVED** [review]

**Comment by lschuermann** [comment]
@bradjc Hm, it seems like the merged result has some issues around macro expansion. Are you able to rebase and perhaps reproduce locally?

**Comment by bradjc** [comment]
Updated a new board.

**Review by lschuermann — APPROVED** [review]

**Pull request merged**